src/factors/PartialPose3.jl

```diff
@@ -59,6 +59,25 @@ function compare(a::PriorPose3ZRP, b::PriorPose3ZRP; tol::Float64=1e-10)
 end
 
 
+"""
+    $(TYPEDEF)
+
+Prior for vector measurements on Pose3.
+"""
+Base.@kwdef struct PriorPose3Ref{T<:SamplableBelief} <: IncrementalInference.AbstractPrior
+    Z::T
+    ref::SVector{3, Float64}
+end
+
+getManifold(::PriorPose3Ref) = TranslationGroup(3)
+
+function (cf::CalcFactor{<:PriorPose3Ref})(p_m, p)
+    n_r = cf.factor.ref
+    nRp = p.x[2]
+    n_m = nRp * p_m
+    return n_r - n_m
```

Comment:
Partials must be indicated, and for gravity should only be pitch and roll.

Reply:
It's not that simple. Pitch and Roll are not on the same manifold as the parameters or measurement. 
I don't know yet how to make it work for the non-parametric solver, that's why no partials.

Reply:
Also, what if the sensor is mounted with a pitch of 90 deg?

Reply:
I think this is where the gradients come in - or info per coord

Reply:
that is why we are starting with gyro only, and we can get to accels once we are ready to work on them.  This can become a rabbit hole of calib, 0vel detect etc.  So I say just hold on accels entirely until we are ready for it.  I need your help on the application examples please.

Reply:
I think results will be bad without absolute information. I've added partials as I see it.

Reply:
yep, agree adding gravity alignment will be good, let's just solve the bigger problem of alignments, rotations, and calibrations first.  Will start using this factor soon after